docs/api.md

```diff
@@ -545,6 +550,53 @@ function App() {
 }
 ```
 
+### `useOutletContext`
+
+<details>
+  <summary>Type declaration</summary>
+
+```tsx
+declare function useOutletContext<
+  Context = unknown
+>(): Context;
+```
+
+</details>
+
+Often parent routes manage state or other values you want shared with child routes. You can create your own [context provider](https://reactjs.org/docs/context.html) if you like, but this is such a common situation that it's built-into `<Outlet />`:
+
+```tsx filename=src/routes/dashboard.tsx lines=[7]
+type ContextType = { user: User | null };
+function Dashboard() {
+  const [user, setUser] = React.useState<User | null>(null);
+
+  return (
+    <div>
+      <h1>Dashboard</h1>
+      <Outlet context={user} />
+    </div>
+  );
+}
+
+export function useUser() {
+  return useOutletContext<ContextType>();
+}
+```
```

Comment:
Let's remove the wrapper here and just do:

```jsx
const { user } = useOutletContext();
```

Also ditch the typescript in the doc, TS users can figure it out, or add a second section in this doc showing the generic usage.

Reply:
I've adjusted things to be better 👍